src/components/document/document.tsx

```diff
@@ -427,6 +453,11 @@ export class DocumentComponent extends BaseComponent<IProps, IState> {
     );
   }
 
+  private handleIdeasButtonClick = () => {
+    const document = this.props.document;
+    logDocumentEvent(LogEventName.REQUEST_IDEA, { document });
```

Comment:
It would be worth putting a comment here pointing people to the exemplar controller rule, since otherwise it will be mysterious to future programmers how this button works!